src/server.c

```diff
@@ -574,6 +574,11 @@ server_finish(struct server *server)
 	}
 	wl_display_destroy_clients(server->wl_display);
 
+	int status;
+	if (primary_client_pid) {
+		waitpid(primary_client_pid, &status, 0);
```

Comment:
This shouldn't be required anymore because we have a global SIGCHLD handler since the pipe menu implementation.